### PR TITLE
KAFKA-17082: replace kafka.utils.LogCaptureAppender with org.apache.kafka.common.utils.LogCaptureAppender

### PR DESCRIPTION
**Pull request by FrankYang0529**
* Reuse `org.apache.kafka.common.utils.LogCaptureAppender`.
* Refactor `AclCommandTest` and `ControllerIntegrationTest`.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/test/scala/unit/kafka/admin/AclCommandTest.scala [hunk 1]
```diff
@@ -217,17 +217,17 @@ class AclCommandTest extends QuorumTestHarness with Logging {
...
-      LogCaptureAppender.setClassLoggerLevel(classOf[AppInfoParser], previousLevel)
-      LogCaptureAppender.unregister(appender)
+      appender.close()
     }
-    val warning = appender.getMessages.find(e => e.getLevel == Level.WARN &&
-      e.getThrowableInformation != null &&
-      e.getThrowableInformation.getThrowable.getClass.getName == classOf[InstanceAlreadyExistsException].getName)
-    assertFalse(warning.isDefined, "There should be no warnings about multiple registration of mbeans")
+    val warning = appender.getEvents.stream.filter(e => e.getLevel == Level.WARN.toString &&
+      e.getThrowableClassName.isPresent &&
```
Comment (chia7712) [inline core/src/test/scala/unit/kafka/admin/AclCommandTest.scala]: `e.getThrowableClassName.filter(_ ==classOf[InstanceAlreadyExistsException].getName).isPresent)`

File: core/src/test/scala/unit/kafka/controller/ControllerIntegrationTest.scala [hunk 2]
```diff
@@ -1881,14 +1882,14 @@ class ControllerIntegrationTest extends QuorumTestHarness {
       TestUtils.waitUntilTrue(() => !controller.isActive, "Controller fails to resign")
 
       // Expect to capture the ControllerMovedException in the log of ControllerEventThread
-      val event = appender.getMessages.find(e => e.getLevel == Level.INFO
-        && e.getThrowableInformation != null
-        && e.getThrowableInformation.getThrowable.getClass.getName.equals(classOf[ControllerMovedException].getName))
-      assertTrue(event.isDefined)
+      val event = appender.getEvents.stream().filter(e => e.getLevel == Level.INFO.toString
+        && e.getThrowableClassName.isPresent
```
Comment (chia7712) [inline core/src/test/scala/unit/kafka/controller/ControllerIntegrationTest.scala]: ditto

**Review by chia7712 — COMMENTED** [review]
@FrankYang0529 nice patch!

File: core/src/test/scala/unit/kafka/admin/AclCommandTest.scala (hunk 1 above)
Comment (FrankYang0529) [reply core/src/test/scala/unit/kafka/admin/AclCommandTest.scala]: Thanks for the suggestion. Updated it.

**Comment by chia7712** [comment]
>  [2024-07-17T12:14:23.174Z] java.io.IOException: No space left on device

@FrankYang0529 please rebase code to trigger QA again. not lucky today :cry

**Comment by FrankYang0529** [comment]
> > [2024-07-17T12:14:23.174Z] java.io.IOException: No space left on device
> 
> @FrankYang0529 please rebase code to trigger QA again. not lucky today :cry

Yes, rebased it. Thank you.

File: clients/src/test/java/org/apache/kafka/common/utils/LogCaptureAppender.java [hunk 3]
```diff
@@ -80,6 +86,17 @@ public static LogCaptureAppender createAndRegister(final Class<?> clazz) {
         return logCaptureAppender;
     }
 
+    public void setClassLoggerToInfo(final Class<?> clazz) {
```
Comment (chia7712) [inline clients/src/test/java/org/apache/kafka/common/utils/LogCaptureAppender.java]: Could we add new method `setClassLogger` to replace those `setClassLoggerToXXX`?
```java
    public void setClassLogger(final Class<?> clazz, Level level) {
        logLevelChanges.add(new LogLevelChange(Logger.getLogger(clazz).getLevel(), clazz));
        Logger.getLogger(clazz).setLevel(level);
    }
```

That is more simple and not verbose to me

**Review by chia7712 — COMMENTED** [review]
@FrankYang0529 thansk for this patch.

File: clients/src/test/java/org/apache/kafka/common/utils/LogCaptureAppender.java (hunk 3 above)
Comment (FrankYang0529) [reply clients/src/test/java/org/apache/kafka/common/utils/LogCaptureAppender.java]: Agree, updated it. Thank you.

File: core/src/test/scala/unit/kafka/admin/AclCommandTest.scala [hunk 4]
```diff
@@ -217,17 +217,16 @@ class AclCommandTest extends QuorumTestHarness with Logging {
...
     } finally {
-      LogCaptureAppender.setClassLoggerLevel(classOf[AppInfoParser], previousLevel)
-      LogCaptureAppender.unregister(appender)
+      appender.close()
     }
-    val warning = appender.getMessages.find(e => e.getLevel == Level.WARN &&
-      e.getThrowableInformation != null &&
-      e.getThrowableInformation.getThrowable.getClass.getName == classOf[InstanceAlreadyExistsException].getName)
-    assertFalse(warning.isDefined, "There should be no warnings about multiple registration of mbeans")
+    val warning = appender.getEvents.stream.filter(e => e.getLevel == Level.WARN.toString &&
```
Comment (chia7712) [inline core/src/test/scala/unit/kafka/admin/AclCommandTest.scala]: Could you please decorate it a bit?
```scala
    assertEquals(0, appender.getEvents.stream
      .filter(e => e.getLevel == Level.WARN.toString)
      .filter(_.getThrowableClassName.filter(_ == classOf[InstanceAlreadyExistsException].getName).isPresent())
      .count(), "There should be no warnings about multiple registration of mbeans")
```

File: core/src/test/scala/unit/kafka/controller/ControllerIntegrationTest.scala [hunk 5]
```diff
@@ -1881,14 +1882,13 @@ class ControllerIntegrationTest extends QuorumTestHarness {
       TestUtils.waitUntilTrue(() => !controller.isActive, "Controller fails to resign")
 
       // Expect to capture the ControllerMovedException in the log of ControllerEventThread
-      val event = appender.getMessages.find(e => e.getLevel == Level.INFO
-        && e.getThrowableInformation != null
-        && e.getThrowableInformation.getThrowable.getClass.getName.equals(classOf[ControllerMovedException].getName))
-      assertTrue(event.isDefined)
+      val event = appender.getEvents.stream().filter(e => e.getLevel == Level.INFO.toString
```
Comment (chia7712) [inline core/src/test/scala/unit/kafka/controller/ControllerIntegrationTest.scala]: ditto

**Review by chia7712 — COMMENTED** [review]
@FrankYang0529 thanks for updated patch

File: core/src/test/scala/unit/kafka/admin/AclCommandTest.scala (hunk 4 above)
Comment (FrankYang0529) [reply core/src/test/scala/unit/kafka/admin/AclCommandTest.scala]: Thanks for the suggestion. Update it!

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request merged**